apps/quick-learn-backend/src/routes/course/course.service.ts

```diff
@@ -267,4 +267,15 @@ export class CourseService extends BasicCrudService<CourseEntity> {
       limit,
     };
   }
+
+  async deleteCourse(id: number): Promise<void> {
```

Comment:
Also, handle the deletion of the lessons which are related to the course.

Reply:
Looking at the lesson entity, I notice that we have onDelete: 'CASCADE' set on the course relationship, which means when a course is deleted, all its associated lessons will be automatically deleted by the database.